java/ffi/pom.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Licensed to the Apache Software Foundation (ASF) under one or more contributor
+  license agreements. See the NOTICE file distributed with this work for additional
+  information regarding copyright ownership. The ASF licenses this file to
+  You under the Apache License, Version 2.0 (the "License"); you may not use
+  this file except in compliance with the License. You may obtain a copy of
+  the License at http://www.apache.org/licenses/LICENSE-2.0 Unless required
+  by applicable law or agreed to in writing, software distributed under the
+  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+  OF ANY KIND, either express or implied. See the License for the specific
+  language governing permissions and limitations under the License. -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>arrow-java-root</artifactId>
+        <groupId>org.apache.arrow</groupId>
+        <version>6.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>arrow-ffi</artifactId>
```

Comment:
sorry I am a little curious what does `ffi` stand for?

Reply:
[Foreign function interface](https://en.wikipedia.org/wiki/Foreign_function_interface)

Reply:
I have been considering if `abi` could be able to describe what happens there better than `ffi`. I am not sure but in Java it seems that when it comes to `foreign` it might implie something related to [project panama](https://openjdk.java.net/projects/panama/)

Reply:
Another naming candidate is bridge (from the cpp implementation). I will change to whatever is chosen. 

Personally I don't associate FFI with project panama, it's a well known term way before project panama started. But I understand your point.

Reply:
Yep the naming doesn't seem to be blocking anyway. Though I was thinking that we might be interested in integrating functionalities from Panama into Arrow once we decide to upgrade to higher Java version (currently we are on 1.8). At that time it might be clearer to have distinction between the module `ffi` or other parts using Panama FFI. Another point is that we already have other modules using JNI which is also literally just a form of FFI. I think if we are introducing C data interface we may want users aim to **data** rather than **function**. Anyway, didn't want to block anything, just my 2 cents.